src/text-editor.coffee

```diff
@@ -818,114 +818,133 @@ class TextEditor extends Model
       @transact groupingInterval, =>
         fn(selection, index) for selection, index in @getSelectionsOrderedByBufferPosition()
 
-  # Move lines intersection the most recent selection up by one row in screen
+  # Move lines intersection the most recent selection or multiple selections up by one row in screen
   # coordinates.
   moveLineUp: ->
-    selection = @getSelectedBufferRange()
-    return if selection.start.row is 0
-    lastRow = @buffer.getLastRow()
-    return if selection.isEmpty() and selection.start.row is lastRow and @buffer.getLastLine() is ''
+    selections = @getSelectionsOrderedByBufferPosition()
```

Comment:
Sorry, but I realized when I woke up this morning that we probably don't actually have to refer to the selection objects directly... I had changed it so we could reassign the selection ranges one at a time, but that had bad interactions with selection merging, so I went back to your approach of just resetting the selected buffer ranges at the end. So you can probably switch back to your original approach of just working with the sorted ranges.